File: 1299/solution.ts
function replaceElements(arr: number[]): number[] {
  let num = -1;
  for (let i = arr.length - 1; i >= 0; i--) {
    [arr[i], num] = [num, Math.max(num, arr[i])];
  }
  return arr;
}

File: 1299/solution.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int right = -1;
        for (int i = arr.length - 1; i >= 0; i--) {
            int biggest = Math.max(right, arr[i]);
            arr[i] = right;
            right = biggest;
        }
        return arr;
    }
}